include/kernel/thread.h

```diff
@@ -309,3 +285,28 @@ void awaken(void);
 void postpone(sched_context_t *sc);
 #endif
 
+/* Update the kernels timestamp and stores in ksCurTime.
+ * The difference between the previous kernel timestamp and the one just read
+ * is stored in ksConsumed.
+ *
+ * Should be called on every kernel entry where threads can be billed.
+ * May be called without holding the kernel lock!
+ */
+static inline void updateTimestamp(void)
+{
+#ifdef CONFIG_KERNEL_MCS
+    time_t prev = NODE_STATE(ksCurTime);
+    NODE_STATE(ksCurTime) = getCurrentTime();
```

Comment:
I wonder, should we read this into a local variable and then have just one write back to memory? I'd trust the compiler to figure out it can do thus, but doing it explicitly in the source might play save and clearly state we want this.